config/validator/service/proposer_settings.go

```diff
@@ -136,6 +147,35 @@ type ProposerOption struct {
 	BuilderConfig      *BuilderConfig
 }
 
+// Clone creates a deep copy of proposer option
+func (po *ProposerOption) Clone() *ProposerOption {
+	if po == nil {
+		return nil
+	}
+	p := &ProposerOption{}
+	if po.FeeRecipientConfig != nil {
+		p.FeeRecipientConfig = po.FeeRecipientConfig.Clone()
+	}
+	if po.BuilderConfig != nil {
+		p.BuilderConfig = po.BuilderConfig.Clone()
+	}
+	return p
+}
+
+func (po *ProposerOption) ToConsensus() *validatorpb.ProposerOptionPayload {
+	if po == nil {
```

Comment:
Is it even possible to enter into this function if `po == nil`?

Reply:
happens on key manager APIs

Reply:
You'r right, I just learned in go it was possible to do:
```go
f.bar()
```

even i `f` is `nil`.

Like:

![image](https://github.com/prysmaticlabs/prysm/assets/4943830/84ae35e5-299f-4c8f-80cb-b11b99e5e858)

==> Should we add receiver nilness check in each method?